components/nexus-bootstrap/src/main/java/org/sonatype/nexus/bootstrap/jetty/ConnectorManager.java

```diff
@@ -0,0 +1,157 @@
+ */
+public class ConnectorManager
+{
+  private final Server server;
+
+  private final Map<String, String> properties;
+
+  private final Map<String, Object> namedBeans;
+
+  private final IdentityHashMap<ConnectorConfiguration, ServerConnector> managedConnectors;
+
+  public ConnectorManager(final Server server,
+                          final Map<String, String> properties,
+                          final Map<String, Object> namedBeans)
+  {
+    this.server = server;
+    this.properties = properties;
+    this.namedBeans = namedBeans;
+    this.managedConnectors = new IdentityHashMap<>();
+  }
+
+  public ServerConnector addConnector(final ConnectorConfiguration connectorConfiguration)
+  {
+    final HttpConfiguration httpConfiguration =
+        connectorConfiguration.customize(defaultHttpConfiguration(connectorConfiguration.getScheme()));
+
+    final ServerConnector serverConnector;
+    if (HttpScheme.HTTP == connectorConfiguration.getScheme()) {
+      final ConnectionFactory connectionFactory = new InstrumentedConnectionFactory(
+          new HttpConnectionFactory(httpConfiguration)
+      );
+      serverConnector = new ServerConnector(
+          server,
+          Integer.valueOf(property("http.acceptors", "-1")),
+          Integer.valueOf(property("http.selectors", "-1")),
+          connectionFactory
+      );
+    }
+    else if (HttpScheme.HTTPS == connectorConfiguration.getScheme()) {
+      verifyHttpsEnabled();
+      final SslContextFactory sslContextFactory = bean("sslContextFactory", SslContextFactory.class);
+      final ConnectionFactory connectionFactory = new InstrumentedConnectionFactory(
+          new SslConnectionFactory(sslContextFactory, "http/1.1")
+      );
+      serverConnector = new ServerConnector(
+          server,
+          Integer.valueOf(property("ssl.acceptors", "-1")),
+          Integer.valueOf(property("ssl.selectors", "-1")),
+          connectionFactory,
+          new HttpConnectionFactory(httpConfiguration)
+      );
+    }
+    else {
+      throw new IllegalArgumentException("Unsupported HTTP Scheme: " + connectorConfiguration.getScheme());
+    }
+    serverConnector.setHost(property("application-host", null));
+    serverConnector.setPort(connectorConfiguration.getPort());
+    serverConnector.setIdleTimeout(Long.valueOf(property("http.tineout", "30000")));
+    serverConnector.setSoLingerTime(Integer.valueOf(property("http.soLingerTime", "-1")));
+    serverConnector.setAcceptorPriorityDelta(Integer.valueOf(property("http.acceptorPriorityDelta", "0")));
+    serverConnector.setSelectorPriorityDelta(Integer.valueOf(property("http.selectorPriorityDelta", "0")));
+    serverConnector.setAcceptQueueSize(Integer.valueOf(property("http.acceptQueueSize", "0")));
+
+    managedConnectors.put(connectorConfiguration, serverConnector);
+    server.addConnector(serverConnector);
+    try {
+      serverConnector.start();
+    }
+    catch (Exception e) {
+      throw new RuntimeException(e);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Define and throw a dedicated exception instead of using a generic one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS00112)